### ISSUE TITLE
Provide a way to allow for third-party extension without making the pointer public. This could look something like this:

### ISSUE THREAD

**Issue by Steven0351** [SwiftyTesseract/SwiftyTesseract#69]
```swift
public func withPointer<A>(_ fn: (TessBaseAPI) -> A) -> A {
  fn(tesseract)
}
```

**Comment by Steven0351** [comment]
This has been implemented in the Swift 5.3 branch but has net yet been merged into develop

**Comment by Steven0351** [comment]
#74
